File: backend/src/main/java/tom/projects/monkeylog/model/user/UserSettings.java
package tom.projects.monkeylog.model.user;

import jakarta.persistence.Entity;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.Id;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import org.hibernate.annotations.UuidGenerator;

import java.util.UUID;

@Entity
@Getter
@Setter
@AllArgsConstructor
@NoArgsConstructor
public class UserSettings implements UserOwned {
    @Id
    @GeneratedValue
    @UuidGenerator
    private UUID id;
    private UUID userId;
    private String metric;
    private String theme;
}

File: backend/src/main/kotlin/tom/com/monkeylog/model/user/UserSettings.kt
package tom.com.monkeylog.model.user

import jakarta.persistence.Entity
import jakarta.persistence.GeneratedValue
import jakarta.persistence.Id
import org.hibernate.annotations.UuidGenerator
import java.util.*

@Entity
class UserSettings(
    @Id
    @GeneratedValue
    @UuidGenerator
    var id: UUID? = null,
    var metric: String,
    var theme: String,
    override var userId: UUID? = null,
) : UserOwned
